### PR TITLE
#13144: Add documentation for tensor creation ops, matmul ops

### PR DESCRIPTION
**Pull request by mcw-anasuya**
### Ticket
Link to Github Issue: #13144

### What's changed
- Updated description according to PyTorch Reference
- Added examples to ops
- Fixed additional bugs

### List of Ops
- [x] ttnn.arange
- [x] ttnn.empty
- [x] ttnn.empty_like
- [x] ttnn.zeros
- [x] ttnn.zeros_like
- [x] ttnn.ones
- [x] ttnn.ones_like
- [x] ttnn.full
- [x] ttnn.full_like
- [x] ttnn.matmul
- [x] ttnn.linear

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/11051499233) - PASSED

### CODE REVIEW

File: ttnn/cpp/pybind11/operations/creation.hpp [hunk 1]
```diff
@@ -84,22 +92,31 @@ void bind_full_operation(py::module& module, const creation_operation_t& operati
 }
 
 template <typename creation_operation_t>
-void bind_full_operation_with_hard_coded_value(py::module& module, const creation_operation_t& operation) {
+void bind_full_operation_with_hard_coded_value(py::module& module, const creation_operation_t& operation, const std::string& value_string) {
     auto doc = fmt::format(
         R"doc(
-        Creates a tensor with the specified shape and fills it with the value of {0}.
+        Creates a tensor with the specified shape and fills it with the value of {1}.
```
Comment (Aswinmcw) [inline ttnn/cpp/pybind11/operations/creation.hpp:98]: this shouldn't be 2 instead of 1?

**Review by VirdhatchaniKN — COMMENTED** [review]
Can you update the post-commit CI test link in the PR description ?

**Review by VirdhatchaniKN — APPROVED** [review]

**Review by bharane-ab — APPROVED** [review]

**Review by yugaoTT — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/matmul/matmul_pybind.cpp [hunk 2]
```diff
@@ -158,8 +158,8 @@ void py_module(py::module& module) {
         If the input tensors have more than two dimensions, the additional, front,
         dimensions may be used for batched matrix multiply.
         These front dimensions may also be referred to as batch dimensions.
-        E.g. a tensor with dimensions :math:`(a \\times b \\times c \\times d)`
-        has batch dimensions a and b.
+        E.g. a tensor with dimensions (`a` x `b` x `c` x `d`)
+        has batch dimensions `a` and `b`.
         The following are the allowed possibilities for batch dimensions.
         Examples below show concrete operations and tensor sizes.
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/matmul/matmul_pybind.cpp:165]: For line 166, please change "If all batch dimensions are all of size" to "If all batch dimensions are of size"

**Review by bbradelTT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by mouliraj-mcw — APPROVED** [review]

**Review by bbradelTT — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

File: ttnn/cpp/pybind11/operations/creation.hpp [hunk 3]
```diff
@@ -296,13 +351,33 @@ void bind_empty_operation(py::module& module) {
             py::arg("shape"),
             py::arg("dtype") = DataType::BFLOAT16,
             py::arg("layout") = Layout::ROW_MAJOR,
-            py::arg("device") = nullptr,
+            py::arg("device"),
```
Comment (cfjchu) [inline ttnn/cpp/pybind11/operations/creation.hpp:354]: does this compile? args with defaults are supposed to come after?

**Pull request merged**